arangod/Agency/Supervision.cpp

```diff
@@ -1276,7 +1279,101 @@ void Supervision::workJobs() {
+  return !rebootID.second || rebootID.first != wantedRebootID;
+}
+
+void Supervision::deleteBrokenDatabase(std::string const& database, std::string const& coordinatorID, uint64_t rebootID) {
+  auto envelope = std::make_shared<Builder>();
+  {
+    VPackArrayBuilder trxs(envelope.get());
+    {
+      VPackArrayBuilder trx(envelope.get());
+      {
+        VPackObjectBuilder operation(envelope.get());
+
+        // increment Plan Version
+        {
+          VPackObjectBuilder o(envelope.get(), _agencyPrefix + "/" + PLAN_VERSION);
+          envelope->add("op", VPackValue("increment"));
+        }
 
+        // delete the database from Plan/Databases
+        {
+          VPackObjectBuilder o(envelope.get(), _agencyPrefix + planDBPrefix + database);
+          envelope->add("op", VPackValue("delete"));
+        }
+
+        // delete the database from Plan/Collections
+        {
+          VPackObjectBuilder o(envelope.get(), _agencyPrefix + planColPrefix + database);
+          envelope->add("op", VPackValue("delete"));
+        }
+      }
+      {
+        // precondition that this database is still in Plan and is building
+        VPackObjectBuilder precondition(envelope.get());
+        envelope->add(_agencyPrefix + planDBPrefix + database + "/" + StaticStrings::DatabaseIsBuilding, VPackValue(true));
+        envelope->add(_agencyPrefix + planDBPrefix + database + "/" + StaticStrings::DatabaseCoordinatorRebootId, VPackValue(rebootID));
+        envelope->add(_agencyPrefix + planDBPrefix + database + "/" + StaticStrings::DatabaseCoordinator, VPackValue(coordinatorID));
+      }
+    }
+  }
+
+
+  write_ret_t res = _agent->write(envelope);
+  if (!res.successful()) {
+    LOG_TOPIC("38482", DEBUG, Logger::SUPERVISION)
+        << "failed to delete broken database in agency. Will retry.";
+  }
+}
+
+void Supervision::checkBrokenCreatedDatabases() {
+  _lock.assertLockedByCurrentThread();
+
+  // check if snapshot has databases
+  std::pair<Node const&, bool> databases = _snapshot.hasAsNode(planDBPrefix);
+  if (!databases.second) {
+    return;
+  }
+
+  // dbpair is <std::string, std::shared_ptr<Node>>
+  for (auto const& dbpair : databases.first.children()) {
+    std::shared_ptr<Node> const& db = dbpair.second;
+
+    // check if isBuilding is set and it is true
+    std::pair<bool, bool> isBuilding = db->hasAsBool(StaticStrings::DatabaseIsBuilding);
+    if (isBuilding.first && isBuilding.second) {
+
+      // this database is currently beeing built
+      //  check if the coordinator exists and its reboot is the same as specified
+      std::pair<uint64_t, bool> rebootID = db->hasAsUInt(StaticStrings::DatabaseCoordinatorRebootId);
+      std::pair<std::string, bool> coordinatorID = db->hasAsString(StaticStrings::DatabaseCoordinator);
+
+      if (!rebootID.second || !coordinatorID.second) {
+        // incomplete data, should not happen
+        continue;   // continue or delete?
```

Comment:
I think here print a warning, and delete the DB. It should (as per specification) never happen that a database has `isBuilding` but no `rebootId` or `coordinatorId`